go/tools/releaser/releaser.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// releaser is a tool for maintaining rules_go and Gazelle. It automates
+// multiple tasks related to preparing releases, like upgrading dependencies,
+// and uploading release artifacts.
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+func main() {
+	if err := run(context.Background(), os.Stderr, os.Args[1:]); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+type command struct {
+	name, description, help string
+	run                     func(context.Context, io.Writer, []string) error
+}
+
+var commands []command
+
+// break init cycle
+func init() {
+	commands = []command{
+		helpCmd,
+		upgradeDep,
+	}
+}
+
+func run(ctx context.Context, stderr io.Writer, args []string) error {
+	if len(args) == 0 {
+		return errors.New("no command specified. For a list of commands, run:\n\treleaser help")
+	}
+	name, args := args[0], args[1:]
+	if name == "-h" || name == "-help" || name == "--help" {
```

Comment:
I think you should iterate through args and do this check so it doesn't have to be first

```go
for _, arg := range args {
  if arg == "-h" || arg == "-help" || arg == "--help" {
    return helpCmd.run(ctx, stderr, args)
  }
}
```